### PR TITLE
docs(gradle): add gradle overview page

### PR DESCRIPTION
**Pull request by xiongemi**
## Expected Behavior

https://nx-31or2tbkt-nrwl.vercel.app/nx-api/gradle

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by leosvelperez — APPROVED** [review]

File: docs/shared/packages/gradle/gradle-plugin.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Overview of the Nx Gradle Plugin
+description: The Nx Gradle plugin contains generators and plugins that are useful for Gradle projects in an Nx workspace.
+---
+
+The Nx Gradle plugin contains generators and plugins that are useful for Gradle projects in an Nx workspace.
```
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:6]: Please add a quick description of Gradle. Most other plugin overview pages also give a quick description of the underlying tool (Gradle).

I also think the main highlight at the moment is the plugin which links gradle tasks to Nx tasks, not generators.

https://nx-dev-git-fork-xiongemi-feature-nxc-544-add-gradle-ce7275-nrwl.vercel.app/nx-api/jest/documents/overview

File: docs/shared/packages/gradle/gradle-plugin.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+### Installation
+
+{% callout type="note" title="Keep Nx Package Versions In Sync" %}
+Make sure to install the `@nx/gradle` version that matches the version of `nx` in your repository. If the version numbers get out of sync, you can encounter some difficult to debug errors. You can [fix Nx version mismatches with this recipe](/recipes/tips-n-tricks/keep-nx-versions-in-sync).
+{% /callout %}
+
+In any Nx workspace, you can install `@nx/gradle` by running the following command:
+
+```shell {% skipRescope=true %}
+nx init
```
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:19]: This does not match the sentence above. It's already a Nx workspace.

File: docs/shared/packages/gradle/gradle-plugin.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+nx add @nx/gradle
+```
+
+### How @nx/gradle Infers Tasks
+
+The `@nx/gradle` plugin will create a task for any project that has an app configuration file present. Any of the following files will be recognized as an app configuration file:
```
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:25]:
```suggestion
The `@nx/gradle` plugin will create an Nx project for each Gradle configuration file present. Any of the following files will be recognized as a Gradle configuration file:
```

File: docs/shared/packages/gradle/gradle-plugin.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+- `gradle.build`
+- `gradle.build.kts`
+
+### View Inferred Tasks
+
+To view inferred tasks for a project, open the [project details view](/concepts/inferred-tasks) in Nx Console or run `nx show project my-project --web` in the command line.
```
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:32]:
```suggestion
To view inferred tasks for a project, open the [project details view](/features/explore-graph#explore-projects-in-your-workspace) in Nx Console or run `nx show project my-project --web` in the command line.
```

File: docs/shared/packages/gradle/gradle-plugin.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+---
+title: Overview of the Nx Gradle Plugin
+description: The Nx Gradle plugin contains generators and plugins that are useful for Gradle projects in an Nx workspace.
+---
+
+The Nx Gradle plugin contains generators and plugins that are useful for Gradle projects in an Nx workspace.
+
+## Setting Up @nx/gradle
```
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:8]: We should have a section here about setting up a new workspace with gradle + Nx. Like the Nest one: https://nx-dev-git-fork-xiongemi-feature-nxc-544-add-gradle-ce7275-nrwl.vercel.app/nx-api/next/documents/overview

We should have 2 versions:
* Gradle Init
* Spring Initializr

File: docs/shared/packages/gradle/gradle-plugin.md (hunk 1 above)
Comment (FrozenPandaz) [inline docs/shared/packages/gradle/gradle-plugin.md:6]: I also think we need a section on why one would use Nx and Gradle together.

It should include information such as:
* Being able to use the Nx cache on top of the Gradle cache to speed up tasks
* Being able to distribute gradle tasks via Nx Cloud
* Being able to view the project graph via Nx including relationships to non-Gradle projects

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: docs/shared/packages/gradle/gradle-plugin.md (hunk 4 above)
Comment (xiongemi) [reply docs/shared/packages/gradle/gradle-plugin.md]: changed

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**